### ISSUE TITLE
Review: MoreEnumerable.RandomSubset()

### ISSUE THREAD

**Issue by GoogleCodeExporter** [blyzer/morelinq#39]
```
Purpose of code changes on this branch:

Rev 123: Implementation of the RandomSubset extension method which returns
a sequence of randomly selected elements from the original sequence. This
operator is deferred but it is NOT streaming. This operator performs a
random shuffle of the original elements, from which it then selects the
first N elements. As an optimization, only a partial shuffle is performed -
instead of a complete shuffle.

When reviewing my code changes, please focus on:

* The public interface of the extension method.
* The effectiveness and clarity of the available XML comment documentation.
* How well this operator fits into the MoreLINQ ecosystem of
extension methods.
* The extent of coverage available from the corresponding unit tests.
* Whether edge cases are correctly identified and handled in the
implementation and tests.
* The potential performance implications of the selected algorithm.

After the review, I'll merge this branch into:
/trunk


```

Original issue reported on code.google.com by `ambientl...@gmail.com` on 17 Jan 2010 at 5:31

**Comment by GoogleCodeExporter** [comment]
```
Now migrated to Hg and available in clone:
http://code.google.com/r/azizatif-morelinq-evenmore/
This clone is ready to be cloned :O) for further review. It also has the 
main/default branch already merged in to bring it up to date.
```

Original comment by `azizatif` on 25 May 2012 at 11:38

**Comment by GoogleCodeExporter** [comment]
```
Will this be available in the NuGet package at some point in the near future?
```

Original comment by `DHilgart...@gmail.com` on 5 Jun 2013 at 11:13

**Comment by GoogleCodeExporter** [comment]
Original comment by `azizatif` on 11 Jun 2013 at 8:14
- Changed state: **Started**

**Comment by GoogleCodeExporter** [comment]
```
Following no response from ambientlion (who possibly/assuming lost interest 
given it has taken years to get this far), taking ownership of issue to 
consider as addition for MoreLINQ 2.0 milestone.
```

Original comment by `azizatif` on 12 Jun 2013 at 10:10
- Added labels: **Component-Logic**, **Milestone-Release2.0**, **Priority-High**, **Type-Enhancement**
- Removed labels: **Priority-Medium**, **Type-Review**

**Comment by GoogleCodeExporter** [comment]
```
This issue was closed by revision ecbb044b3f86.
```

Original comment by `azizatif` on 14 Jun 2013 at 5:18
- Changed state: **Fixed**
